standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -8582,7 +8582,8 @@ public GetOpenTxnsInfoResponse get_open_txns_info() throws TException {
   public OpenTxnsResponse open_txns(OpenTxnRequest rqst) throws TException {
     OpenTxnsResponse response = getTxnHandler().openTxns(rqst);
     List<Long> txnIds = response.getTxn_ids();
-    if (txnIds != null && listeners != null && !listeners.isEmpty()) {
+    boolean isHiveReplTxn = rqst.isSetReplPolicy() && rqst.getTxn_type() == TxnType.DEFAULT;
```

Comment:
this conditional have been copy pasted to a few places - can we move it into a method?

Reply:
rqst variable can be of different types (OpenTxnRequest/ CommitTxnRequest/ AbortTxnRequest). So, this statement may not be generalised.